tests/test_utils_http.py

```diff
@@ -311,3 +312,23 @@ def _is_uuid(string: str) -> bool:
 )
 def test_fix_hf_endpoint_in_url(base_url: str, endpoint: Optional[str], expected_url: str) -> None:
     assert fix_hf_endpoint_in_url(base_url, endpoint) == expected_url
+
+
+def test_adjust_range_header():
+    # Basic cases
+    assert _adjust_range_header(None, 10) == "bytes=10-"
+    assert _adjust_range_header("bytes=0-100", 10) == "bytes=10-100"
+    assert _adjust_range_header("bytes=-100", 10) == "bytes=-90"
+
+    with pytest.raises(RuntimeError):
+        _adjust_range_header("invalid", 10)
+
+    with pytest.raises(RuntimeError):
+        _adjust_range_header("bytes=-", 10)
+
+    # Multiple ranges
+    assert _adjust_range_header("bytes=0-100,200-300", 10) == "bytes=10-"
```

Comment:
I think it's better to raise also in that case i.e. we don't want to support multi-ranges.
And if in the future we want to support them, then we'll see what we want to do.

Reply:
Hmm, not sure which one ('give a warning' or 'raise error') is better, any suggestions? @hanouticelina

Reply:
The advantage of raising an error is that we won't introduce an a breaking change in the future.
If we simply warn and then realize it's not enough, switching to "raise" would be a breaking change. And since it's only an internal method used in a constrained case, I do think it's better to support only the explicit cases we require.